Dockerfile

```diff
@@ -24,11 +24,16 @@ RUN wget -qO- https://packages.microsoft.com/keys/microsoft.asc | gpg --dearmor
 	chown root:root /etc/apt/sources.list.d/microsoft-prod.list
 
 RUN npm install -g serverless
-RUN apt-get update && apt-get install -y azure-cli dotnet-sdk-3.1 maven azure-functions-core-tools
+RUN npm install -g azure-functions-core-tools@3
+RUN apt-get update
+RUN apt install software-properties-common -y
+RUN add-apt-repository ppa:openjdk-r/ppa -y
+RUN apt-get install azure-cli git curl wget gcc zip unzip openjdk-8-jdk maven dotnet-sdk-3.1 python3 python3-pip -y --fix-missing
```

Comment:
Can you move as many of the libraries installed here as possible to the first apt-get on line 11.  
Also note to remove duplicates (curl, wget)